test/cases/schema_statements_test.rb

```diff
@@ -6,7 +6,7 @@ module PostGIS
   class SchemaStatementsTest < ActiveSupport::TestCase
     def test_initialize_type_map
       SpatialModel.with_connection do |connection|
-        assert connection.connected?
+        connection.connect!
```

Comment:
I guess the reasoning here is that `connect!` would raise if not able to connect? Do you have any idea why this has changed though?

Reply:
This was added in #397 , and there is a comment about it: https://github.com/rgeo/activerecord-postgis-adapter/pull/397/files#r1646652915

I think this line was added to highlight why the next command was failing, maybe to try pinpoint the reason the connection sometimes is not there.
I found that this is also dependent of the order the tests run. If you run this test in isolation it would aways fail. If this test run after a test using the connection (eg: `model.connection.create_table`), then this test would pass.